### ISSUE TITLE
Server restart in v1.10.0 - regression?

### ISSUE THREAD

**Issue by hugebdu** [grpc/grpc-node#2668]
### Problem description
WIth version 1.10.0 server cannot be restarted. That wasn't the case with previous versions.

### Reproduction steps
__pseudocode__
```
const server = // init server
server.bindAsync(callback);
server.tryShutdown(callback);
server.bindAsync(callback); // fails with Error('bindAsync called after shutdown')
```

**Comment by murgatroid99** [comment]
I can change this, but depending on what exactly you're trying to accomplish here, you may get a better effect from a different server API.

**Comment by hugebdu** [comment]
My use case with restarts is a test stub — I test some logic (integration test) that is dependent on availability of another gRPC endpoint.

**Comment by murgatroid99** [comment]
OK, I think you would find the new `unbind` API useful. You can unbind the specific port that you want to make unavailable, and then bind it again later.

**Comment by hugebdu** [comment]
Yes, that's what I ended up using. Just thought to report is as a possible regression.
Thanks
